gcc/rust/expand/rust-proc-macro.cc

```diff
@@ -0,0 +1,58 @@
+// This file is part of GCC.
+
+// GCC is free software; you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 3, or (at your option) any later
+// version.
+
+// GCC is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+// for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with GCC; see the file COPYING3.  If not see
+// <http://www.gnu.org/licenses/>.
+
+#include "rust-diagnostics.h"
+#include "rust-proc-macro.h"
+#include <dlfcn.h>
```

Comment:
Yes, this will cause problems as we are expected to use `LoadLibrary()` and such functions. Should I find a solution in this PR or should we postpone to another one ?

Reply:
I think maybe this PR could have something like `#ifdef WIN64` (I can't remember if that is the correct macro), do not include `dlfcn.h` and do not call into `dlopen/dlsym` and instead `rust_sorry_at`. I can check with @simonpcook if your PR builds on windows too :)

Reply:
> I think maybe this PR could have something like `#ifdef WIN64` (I can't remember if that is the correct macro), do not include `dlfcn.h` and do not call into `dlopen/dlsym` and instead `rust_sorry_at`. I can check with @simonpcook if your PR builds on windows too :)

I believe the macro is `_WIN64` instead. I made the changes.